trace/exporterutil/export_flusher.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterutil
+
+import "go.opencensus.io/trace"
+
+// ExportFlusher extends Exporter to support synchronous flushing of any
+// internal buffered spans. As with most aspects of exporting, it is
+// best-effort only.
+type ExportFlusher interface {
```

Comment:
Why do we need this?

We are supposed to filter if the underlying exporter exports to our layer. If not, it is not our responsibility.

Reply:
So the Flush convention as mentioned above already exists. I am propagating the Flush method to the ExcludeSpansExporter because in frameworks (e.g. what I'm working on for Istio) the framework manages the lifecycle of your exporter and needs to call Flush (if the exporter has a Flush). So we need any intermediate wrapping exporters to propagate the Flush to the underlying exporters.

Reply:
How can't the flush'ing can be operated via the underlying exporter? We cannot add more interfaces to support all additional functionality provided by the implementations of an interface.

If it is truly required, it then should be a part of the trace.Exporter interface. I don't understand why the user of ExcludeSpansExporter cannot pin the underlying exporter and flush through that.

Reply:
I didn't strictly need to name the interface. But I've used it so many times unnamed and the convention is so well established that I believe it is useful to have the interface defined here. 

It isn't strictly required by an exporter, I guess if the exporter didn't have any kind of internal buffering there would be no need to have an explicit flush operation. But even if we thought that it were worth including in the official interface, that would be a backwards incompatible change and so would need to follow the deprecation process.